src/main/java/seedu/address/model/task/Description.java

```diff
@@ -10,15 +10,16 @@
 public class Description implements Comparable<Description> {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Description should not be blank or start with white spaces,"
-                    + " and the length should be less than 256 characters.";
+            "Description should only contain alphanumeric characters, punctuations and spaces, "
+                    + "and should not start with punctuations.\n "
+                    + "Description has a character limit of 1-255 characters.";
 
     /*
-     * The first character of the description must not be a whitespace.
-     * For other characters, they must be printable or whitespaces.
+     * The first character of the description must not be a whitespace or punctuation.
+     * For other characters, they must be alphanumeric or spaces.
      * The maximum number of characters allowed is 255.
      */
-    public static final String VALIDATION_REGEX = "[\\p{Alnum}\\p{Punct}][\\p{Alnum}\\p{Punct}\\s]{0,254}";
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum}\\p{Punct}\\s]{0,254}";
```

Comment:
Would it be necessary to only restrict the first character to alphanumeric characters? I think it might be easier for users to read the user guide, and for their use if we allowed both alphanumeric characters and punctuations actually.

i.e. [G2] Complete Milestone 3

Reply:
i agree with ben's concern on this

Reply:
The reason of designing it this way is because if we don't have any restriction on the first character, the name could be all symbols. In that case our find may not be able to find this task through name/description at all.

Reply:
ohh okok makes sense!! ty!!

Reply:
ohh okok makes sense!! ty!!